ui-v2/app/services/feedback.js

```diff
@@ -2,36 +2,41 @@ import Service, { inject as service } from '@ember/service';
 import { get, set } from '@ember/object';
 import callableType from 'consul-ui/utils/callable-type';
 
+const TYPE_SUCCESS = 'success';
+const TYPE_ERROR = 'error';
+const defaultStatus = function(type, obj) {
+  return type;
+};
 export default Service.extend({
   notify: service('flashMessages'),
   logger: service('logger'),
-  execute: function(handle, success, error, controller) {
+  execute: function(handle, action, status = defaultStatus, controller) {
     set(controller, 'isLoading', true);
-    const displaySuccess = callableType(success);
-    const displayError = callableType(error);
+    const getAction = callableType(action);
+    const getStatus = callableType(status);
     const notify = get(this, 'notify');
     return (
       handle()
-        //TODO: pass this through to display success..
-        .then(() => {
+        //TODO: pass this through to getAction..
+        .then(target => {
           notify.add({
-            type: 'success',
+            type: getStatus(TYPE_SUCCESS),
             // here..
-            message: displaySuccess(),
+            action: getAction(),
           });
         })
         .catch(e => {
           get(this, 'logger').execute(e);
           if (e.name === 'TransitionAborted') {
             notify.add({
-              type: 'success',
+              type: getStatus(TYPE_SUCCESS),
               // and here
-              message: displaySuccess(),
+              action: getAction(),
             });
           } else {
             notify.add({
-              type: 'error',
-              message: displayError(e),
+              type: getStatus(TYPE_ERROR, e),
+              action: getAction(),
```

Comment:
Do you think it makes sense to make more methods here instead of passing `action` around as a string at call sites?

```js
// feedback.js
execute: function(handle, action, status = defaultStatus, controller) {
  // ...
},
create: function(handle, status = defaultStatus, controller) {
  return this.execute(handle, 'create', status, controller);
},
update: function(handle, status = defaultStatus, controller) {
  return this.execute(handle, 'update', status, controller);
}
// ...
}

// some-controller.js
actions: {
  doTheThing: function() {
    get(this, 'feedback').create(() => {
      return promise.then(() => {
        return someStuff();
      });
    });
  }
}
```

Reply:
I think this is what you referred to here https://github.com/hashicorp/consul/pull/4573#discussion_r213485062 ? Lemme know if not.